Makefile.setup.mk

```diff
@@ -0,0 +1,170 @@
+SHELL := /bin/bash
+
+.PHONY: create-kind-cluster
+create-kind-cluster:
+	kind create cluster --name kfp --image kindest/node:v1.29.2 --wait 5m && \
+	kubectl version --client --short
+
+.PHONY: build-images
+build-images:
+	./.github/resources/scripts/build-images.sh
+
+.PHONY: deploy-kfp-tekton
+deploy-kfp-tekton:
+	./.github/resources/scripts/deploy-kfp-tekton.sh
+
+.PHONY: setup-kfp-tekton
+setup-kfp-tekton:
+	$(MAKE) build-images && \
+	$(MAKE) deploy-kfp-tekton
+
+.PHONY: deploy-kfp
+deploy-kfp:
+	./.github/resources/scripts/deploy-kfp.sh
+
+.PHONY: setup-kfp
+setup-kfp:
+	$(MAKE) build-images && \
+	$(MAKE) deploy-kfp
+
+.PHONY: setup-python
+setup-python:
+	python3 -m venv .venv && \
+	source .venv/bin/activate
+
+.PHONY: setup-backend-test
+setup-backend-test:
+	python3 -m venv .venv && \
+	source .venv/bin/activate && \
+	pip install -e sdk/python
+
+.PHONY: setup-backend-visualization-test
+setup-backend-visualization-test:
+	$(MAKE) setup-python
+
+.PHONY: setup-frontend-test
+setup-frontend-test:
+	npm cache clean --force && \
+	cd ./frontend && npm ci
+
+.PHONY: setup-grpc-modules-test
+setup-grpc-modules-test:
+	$(MAKE) setup-python && \
+	sudo apt-get update && \
+	sudo apt-get install protobuf-compiler -y && \
+	pip3 install setuptools && \
+	pip3 freeze && \
+	pip3 install wheel==0.42.0 && \
+	pip install sdk/python && \
+	cd ./api && \
+	make clean python && \
+	cd .. && \
+	python3 -m pip install api/v2alpha1/python && \
+	pip install components/google-cloud && \
+	pip install $(shell grep 'pytest==' sdk/python/requirements-dev.txt) && \
+	pytest ./test/gcpc-tests/run_all_gcpc_modules.py
```

Comment:
this looks like it shouldn't be here, as it's only a setup target and the same command is ran in `test-grpc-modulestest-grpc-modules`